app/scripts/views/sign_up.js

```diff
@@ -51,6 +51,17 @@ function (Cocktail, _, p, BaseView, FormView, Template, Session, AuthErrors,
         return p(false);
       }
 
+      /**
+       * check if there is an account we can suggest for an OAuth flow.
+       */
+      var account = this.user.getChooserAccount();
+      var serviceName = this.relier.get('serviceName');
+
+      if (account.get('email') && serviceName) {
```

Comment:
I'm really uncomfortable with the semantics of users who are sent to the "/signin" page to automatically be sent to the "/signup" page and vice versa.

Similar functionality is needed for the firstrun experience, and I [propose sending the user to `/`](https://github.com/mozilla/fxa-content-server/issues/2101#issuecomment-85555029).

Can someone explain the need for this when we have endpoints that make this decision already?